br/pkg/task/stream.go

```diff
@@ -434,14 +434,14 @@ func (s *streamMgr) backupFullSchemas(ctx context.Context, g glue.Glue) error {
 		m.ClusterVersion = clusterVersion
 	})
 
-	schemas, err := backup.BuildFullSchema(s.mgr.GetStorage(), s.cfg.StartTS)
-	if err != nil {
-		return errors.Trace(err)
-	}
+	schemas := backup.NewBackupSchemas(func(storage kv.Storage, fn func(*model.DBInfo, *model.TableInfo)) error {
+		return backup.BuildFullSchema(storage, s.cfg.StartTS, func(dbInfo *model.DBInfo, tableInfo *model.TableInfo) {
+			fn(dbInfo, tableInfo)
+		})
+	}, 0)
 
-	schemasConcurrency := uint(mathutil.Min(backup.DefaultSchemaConcurrency, schemas.Len()))
 	err = schemas.BackupSchemas(ctx, metaWriter, nil, s.mgr.GetStorage(), nil,
-		s.cfg.StartTS, schemasConcurrency, 0, true, nil)
+		s.cfg.StartTS, backup.DefaultSchemaConcurrency, 0, true, nil)
```

Comment:
Concurrency is different now. Is it intentional?

Reply:
That's because the new `schemas` need to iterates all tables to get the size, which is repeated in this situation. Therefore, we don't get the size of `schemas`.

Besides, we can see the `workerpool` in `schemas.BackupSchemas` only push the `worker` to channel. Only when call `ApplyOnErrorGroup`, it would create a new goroutine. So directly use `backup.DefaultSchemaConcurrency` is somewhat equivalent to `mathutil.Min(backup.DefaultSchemaConcurrency, schemas.Len()))`.